spec/helpers.lua

```diff
@@ -561,16 +563,44 @@ local function tcp_server(port, opts, ...)
         end
 
         line = assert(client:receive())
-        client:send((opts.prefix or "") .. line .. "\n")
+
+        if not handshake_done then
+          assert(line == "\\START")
+          client:send("\\OK\n")
+          handshake_done = true
+        else
+          client:send((opts.prefix or "") .. line .. "\n")
+        end
         client:close()
       end
       server:close()
       return line
     end
   }, port, opts)
 
-  local thr = thread:start(...)
-  ngx.sleep(0.01)
+  local thr = thread:start()
+
+  -- not necessary for correctness because we do the handshake,
+  -- but avoids harmless "connection error" messages in the wait loop
+  -- in case the client is ready before the server below.
+  ngx.sleep(0.001)
+
+  local sock = ngx.socket.tcp()
+  sock:settimeout(0.01)
+  while true do
+    local c, er = sock:connect("localhost", port)
+    print(c, er)
```

Comment:
Is this `print()` intentional?

Reply:
Dammit :laughing: Second time this week! That's it, I'm adding a pre-commit hook!

Reply:
```bash
#!/usr/bin/env bash

files=($(git status --porcelain | grep "^[AM]" | cut -b4-))

lua_files=($(echo "${files[@]}" | tr ' ' '\n' | grep "lua$"))
if [ "${#lua_files[@]}" -gt 0 ]
then
   luacheck -q "${lua_files[@]}"
fi

spec_files=($(echo "${files[@]}" | tr ' ' '\n' | grep "spec/"))

fail=0
function complain() {
   echo -e "\033[1;31m *** $@\033[0m"
   echo
   fail=1
}


if ! [ "$WIP" ]
then
   for file in "${spec_files[@]}"
   do
      grep -w 'print' $file && complain "$file has a print statement!"
      grep -w '#only' $file && complain "$file has a #only comment!"
   done
fi

if [ "$fail" = 1 ]
then
   echo "use 'WIP=1 git commit' to override"
   exit 1
fi

exit 0
```

Reply:
I'd also recommend running `grep -R -w -i -- '--- ONLY' t/`